B1925

jhoblitt opened the pull request.

jhoblitt committed:
fix mangled mariadb/mariadbclient patches

jhoblitt committed:
skip testdata_subaru product (LFS repo)

jhoblitt committed:
bump lsst-product-configs to 1.0.1925

jhoblitt commented:
I'm not sure if `1.0.1925` or `1925` makes more sense as a version string.  It would be convenient to tag `conda-lsst` with the same string.

jhoblitt committed:
healpy requires six

jhoblitt commented:
@mjuric I don't have commit rights to this repo but this PR should be ready to commit.  I'm going to move on to the w_2016_08/b1926 tag tomorrow.

mjuric commented on conda_lsst/recipe_maker.py:
```diff
@@ -84,7 +84,7 @@ def gen_conda_package(self, product, sha, eups_version, giturl, eups_deps, eups_
 		# and .cfg files are.
 		if depends_on_internal:
-			product_configs = 'lsst-product-configs ==0.0.1'
+			product_configs = 'lsst-product-configs ==1.0.1925'
```
Add a key to `config.yaml` to read the version from (so we don't have to touch the code). E.g.

```
internal_product_configs_package: "lsst-product-configs ==1.0.1925"
```

mjuric replied:
Come to think of it, maybe we shouldn't do this as it will be super trivial to update the `lsst-product-config` recipe (the meta.yaml), but forget to update `config.yaml`, and then be confused why the build isn't working.

Instead, it may be better to read the version from `lsst-product-config`'s meta.yaml, e.g., like we do [here](https://github.com/mjuric/conda-lsst/blob/master/conda_lsst/recipe_maker.py#L203).

jhoblitt replied:
I was intending to ask your about this but thoughts about this but decided maybe I should wait until after the JTM.  

My first impression was that it would be better to set it in the config.yaml, in the hopes that some day multiple different bNNNNs could be supported with the same core code and just changing the config.yaml (and possibly other `./etc` data).  Today, I'm wondering if that is bit premature to worry about and it would be more convenient to pickup the version automatically.  Hypothetically, would that interact badly with trying to build ontop of an existing conda channel with multiple version of `lsst-product-config`?

jhoblitt replied:
I've decided to go with the suggestion of fishing the version string out of the static recipe and think I have a working patch.  I will repush after building a larger chuck of our package set.

jhoblitt replied:
repushed with the the proposed patch.

jhoblitt committed:
detect lsst-product-configs version from meta.yaml

rather than hard coding the version string, which may need to be
frequently updated.

mjuric commented on etc/recipes/lsst-product-configs/meta.yaml:
```diff
@@ -6,14 +6,14 @@
 
 source:
-  git_tag: 1.0.0
+  git_tag: *package_version
```
I didn't know you could do this. Cool!

mjuric commented:
Looks great, will merge.
